src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
     total = sum(profile)
     N = length(profile)
-    phase = Array(range(0, stop = N - 1) / N) * 2 * pi
+    phase = range(0, stop = N - 1) / N * 2 * pi
```

Comment:
The `range` gives you a compact and efficient data structure that doesn't need to allocate a vector of `N` elements.  The elements of this data structure are evaluated lazily when you access them

Reply:
Does it also work with the final multiplication? When is the final multiplication done?

Reply:
(PS: now that I think of it, it would probably be better to enclose the final factors in a parenthesis to make the multiplication only once. Or it is optimized out?)

Reply:
All those multiplications and division are very efficient, their computational cost is `O(1)`, i.e. they don't depend on the length `N` of the range.  You can use the [`dump`](https://docs.julialang.org/en/v1/base/io-network/#Base.dump) function to inspect the content of data structure:
```julia
julia> N = 10
10

julia> r = range(0, stop = N - 1)
0:9

julia> dump(r)
UnitRange{Int64}
  start: Int64 0
  stop: Int64 9

julia> dump(r / N)
StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}
  ref: Base.TwicePrecision{Float64}
    hi: Float64 0.0
    lo: Float64 0.0
  step: Base.TwicePrecision{Float64}
    hi: Float64 0.09999999999999964
    lo: Float64 3.552713678800501e-16
  len: Int64 10
  offset: Int64 1

julia> dump(r / N *  2)
StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}
  ref: Base.TwicePrecision{Float64}
    hi: Float64 0.0
    lo: Float64 0.0
  step: Base.TwicePrecision{Float64}
    hi: Float64 0.1999999999999993
    lo: Float64 7.105427357601002e-16
  len: Int64 10
  offset: Int64 1

julia> dump(r / N *  2 * pi)
StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}
  ref: Base.TwicePrecision{Float64}
    hi: Float64 0.0
    lo: Float64 0.0
  step: Base.TwicePrecision{Float64}
    hi: Float64 0.6283185307179586
    lo: Float64 0.0
  len: Int64 10
  offset: Int64 1
```
Basically, all multiplications/divisions simply update the `ref` and `step` fields.